Architecture.md

```diff
@@ -28,7 +28,29 @@
 
 #### Client applications deployment
 
-***TODO: [INFRA] fill for the architecture used to deploy clients' applications***
+Pomegranate, the application and deployment manager, is in charge of monitoring and managing the containers' lifecycle, as well as configuring them.
+
+The [Rust](https://www.rust-lang.org/) programming language is used to develop the service. We used it because
+we wanted an efficient and reliable language that could be trusted with this job thanks to its compiled nature and the restrictions it brings to drive us to write better code,
+and with support for the other technologies we use through the crates ecosystem.
+
+Establishing communication with other services is done via [gRPC](https://grpc.io/docs/what-is-grpc/core-concepts/).
+gRPC imposed itself as the go-to framework for internal APIs thanks to its [high performance](https://www.nexthink.com/blog/comparing-grpc-performance)
```

Comment:
Can you please present the benchmarks? Having a link is good, but a sum up could help the reader understand our point. Also, the sentence is missing a full stop.

Reply:
I have to say, it feels wasteful in my opinion. Perhaps I should just include the performance graph with a small caption ?

Reply:
Including the performance graph seems quite a bit of work to add, since the image needs to be placed in a separate folder in the codebase. I was thinking that maybe a quick overview of the benchmark could give a first opinion on its truthfulness, while still giving some support to your claims.